homeassistant/components/growatt_server/number.py

```diff
@@ -0,0 +1,169 @@
+) -> None:
+    """Set up Growatt number entities."""
+    runtime_data = entry.runtime_data
+
+    entities: list[GrowattNumber] = []
+
+    # Add number entities for each MIN device (only supported with V1 API)
+    for device_coordinator in runtime_data.devices.values():
+        if (
+            device_coordinator.device_type == "min"
+            and device_coordinator.api_version == "v1"
+        ):
+            entities.extend(
+                GrowattNumber(
+                    coordinator=device_coordinator,
+                    description=description,
+                )
+                for description in MIN_NUMBER_TYPES
+            )
+
+    async_add_entities(entities)
+
+
+class GrowattNumber(CoordinatorEntity[GrowattCoordinator], NumberEntity):
+    """Representation of a Growatt number."""
+
+    _attr_has_entity_name = True
+    _attr_entity_category = EntityCategory.CONFIG
+    entity_description: GrowattNumberEntityDescription
+
+    def __init__(
+        self,
+        coordinator: GrowattCoordinator,
+        description: GrowattNumberEntityDescription,
+    ) -> None:
+        """Initialize the number."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.device_id}_{description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.device_id)},
+            manufacturer="Growatt",
+            name=coordinator.device_id,
+        )
+
+    @property
+    def native_value(self) -> int | None:
+        """Return the current value of the number."""
+        value = self.coordinator.data.get(self.entity_description.api_key)
+        if value is None:
+            return None
+        return int(value)
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the value of the number."""
+        # Use write_key if specified, otherwise fall back to api_key
+        parameter_id = (
+            self.entity_description.write_key or self.entity_description.api_key
+        )
+        int_value = int(value)
+
+        try:
+            # Use V1 API to write parameter
+            await self.hass.async_add_executor_job(
+                self.coordinator.api.min_write_parameter,
+                self.coordinator.device_id,
+                parameter_id,
+                int_value,
+            )
+        except GrowattV1ApiError as e:
+            _LOGGER.error("Error while setting parameter: %s", e)
+            raise HomeAssistantError(f"Error while setting parameter: {e}") from e
+
+        # If no exception was raised, the write was successful
+        _LOGGER.debug(
+            "Set parameter %s to %s",
+            parameter_id,
+            value,
+        )
+
+        # Update the value in coordinator data
+        self.coordinator.data[self.entity_description.api_key] = int_value
+        self.async_write_ha_state()
```

Comment:
just as a double check, we can't ask the coordinator to refresh to get new data?